<?xml version="1.0"?>
<launch>
  <!-- Joystick Driver -->
  <node pkg="joy" type="joy_node" name="joy">
    <param name="dev" value="/dev/input/js0"/>
    <param name="autorepeat_rate" value="10"/>
  </node>

  <!-- Joystick Teleop -->
  <!-- <rosparam file="$(find vrx_gazebo)/config/diffdrive.yaml" command="load"/>
  <node pkg="joy_teleop" type="joy_teleop.py" name="joy_teleop">
    <remap from="cmd_vel" to="cmd_vel" />
  </node> -->

  <!-- Parameters -->
  <arg name="boat_type" default="single"/>

  <!-- Single motor boat -->
  <group if="$(eval arg('boat_type') == 'single')">
    <node pkg="auto_drive" type="joy_drive" name="joy_drive_node" output="screen">
      <param name="boat_type" value="single"/>
      <remap from="thrust_cmd" to="/wamv/thrusters/moddle_thrust_cmd"/>
      <remap from="angle_cmd" to="/wamv/thrusters/moddle_thrust_angle"/>
    </node>
  </group>

  <!-- Double motor boat -->
  <group if="$(eval arg('boat_type') == 'double')">
    <node pkg="auto_drive" type="joy_drive" name="joy_drive_node" output="screen">
      <param name="boat_type" value="double"/>
      <remap from="left_cmd" to="/wamv/thrusters/left_thrust_cmd"/>
      <remap from="right_cmd" to="/wamv/thrusters/right_thrust_cmd"/>
      <remap from="left_angle" to="/wamv/thrusters/left_thrust_angle"/>
      <remap from="right_angle" to="/wamv/thrusters/right_thrust_angle"/>
      
    </node>
  </group>
</launch>
